code/processes/gateway.q

```diff
@@ -555,9 +565,19 @@ pc:{
  removeclienthandle[x];
  removeserverhandle[x];}
 
+pgs:{.gw.call,:enlist[x]!enlist y};
+
 // override message handlers
-.z.pc:{x@y; .gw.pc[y]}@[value;`.z.pc;{{[x]}}]
-.z.po:{x@y; .gw.po[y]}@[value;`.z.po;{{[x]}}]
+.z.pc:{x@y;.gw.pc[y]}@[value;`.z.pc;{{[x]}}];
+.z.po:{x@y;.gw.po[y]}@[value;`.z.po;{{[x]}}];
+.z.pg:{
+  .gw.pgs[.z.w;1b];
+  :x@y;
+ }@[value;`.z.pg;{{[x]}}];
+.z.ps:{
+  .gw.pgs[.z.w;0b];
+  :x@y;
+ }@[value;`.z.ps;{{[x]}}];
```

Comment:
Ok I think the logic her is correct but could you just make the formatting consistent with the definitions of .z.po/.z.pc?   Also remove the .z.pg commented out line below